### PR TITLE
docment vision.augment for RandTransform and Item Transforms

### PR DESCRIPTION
**Pull request by dienhoa**
This is a simple PR for docment 2 first tasks in vision.augment [issue 29](https://github.com/muellerzr/fastai-docment-sprint/issues/29) . This is much for me to get used with the workflow and everything is aligned with the style guide

### CODE REVIEW

**Comment by dienhoa** [comment]
Some notes: 

- I don't understand clearly what is the use of `nm` in RandTransform and in discord, Zach said might be it is a bug so I left it blank.
- for `b` in `before_call` and `__call__` I believe it can be anything so not sure we should document it.
- `split_idx` in `before_call` I forgot to document, I will add it later in the next PR

**Comment by muellerzr** [comment]
@dienhoa please add it in this PR, so everything is complete :)

**Comment by marii-moe** [comment]
For `nm` I am guessing it is dead code. 

I checked old version of RandTransform and nothing: https://github.com/fastai/fastai/blame/b1507889980e48dac9a015492bcb1c44da2bd1c2/fastai2/vision/augment.py

Also in fastcore: https://github.com/fastai/fastcore/blame/463e76827090b4607ff8d95ca93cb98af31c2682/fastcore/transform.py

Nothing back to the beginning of time seems to be suggesting nm is there, the history for when it was originally added is missing. I do not think it was being used at the beginning of time either.

File: fastai/vision/augment.py
Comment (warner-benjamin) [inline fastai/vision/augment.py:38]: These `split_idx` are a bit tricky, but this one doesn't change whether the transform is applied
```suggestion
        split_idx:int, # Index of dataset when called
```

File: fastai/vision/augment.py
Comment (warner-benjamin) [inline fastai/vision/augment.py:85]: I believe this is a better description
```suggestion
    k:int, # Dihedral transformation to apply
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
I requested some changes. I might have a couple more later.

We are currently discussing what to do about transform's `p`, whether they need docmenting or not, or if the argument name should be changed. Will update when that's resolved.

File: fastai/vision/augment.py
Comment (marii-moe) [reply fastai/vision/augment.py:28]: I think the before call behavior is clear from the name. The purpose of this is that it is done before call which is done before encodes. This is a very convenient method to do batch level preocessing before we ever get to `encodes` at the item level. This is also before type dispatch is applied so we have access to all of the data being passed through the pipeline. 

So I think it would be good to highlight it is a function is applied to the entire batch, including what would not normally be accessible by type dispatch.

File: fastai/vision/augment.py
Comment (dienhoa) [reply fastai/vision/augment.py:38]: @warner-benjamin Thanks. I was thinking too much about it and forgot that it has no effect for transform in `before_call`. However, as you said, `split_idx` is quite tricky so I think maybe we need a new design for it. It's hard to guess that changing the split_idx can have an effect on the Transform

**Comment by dienhoa** [comment]
I've just updated with a new commit. One note: function `crop_pad` has a resize feature which I think is not obvious. Maybe we need to rename it to something like `crop_pad_resize`?

File: fastai/vision/augment.py
Comment (warner-benjamin) [inline fastai/vision/augment.py:43]:
```suggestion
        split_idx:int=None, # Index of the train/valid dataset
```

File: fastai/vision/augment.py
Comment (warner-benjamin) [inline fastai/vision/augment.py:36]:
```suggestion
        split_idx:int, # Index of the train/valid dataset
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]

**Comment by marii-moe** [comment]
@dienhoa `self` should be on the same line as `def before_call` for example. As in benjamin's above suggestion.

**Comment by dienhoa** [comment]
Thanks @marii-moe . If so, I think I fix all the new lines pbs in my latest commit after his suggestion.

I was thinking about something else other than his suggestion

**Review by warner-benjamin — APPROVED** [review]
Looks good. Waiting for CI to run before merging.

File: fastai/vision/augment.py
Comment (marii-moe) [inline fastai/vision/augment.py:38]: Please add Something like "This can be function overridden. Set `self.do` based on `self.p`"

This was brought to my attention by Zach

**Review by marii-moe — CHANGES_REQUESTED** [review]
Sorry about this, only 1 more request.

File: fastai/vision/augment.py
Comment (dienhoa) [reply fastai/vision/augment.py]: Add it. I totally agree with this. In my first iteration, I was thinking that it's some kind of `set_up function` that we can preprocess some attributes like self.do, self.p, self.split_idx, ...

**Pull request merged**
